internal/logger/target/http/http.go

```diff
@@ -121,54 +124,62 @@ func acceptedResponseStatusCode(code int) bool {
 }
 
 func (h *Target) startHTTPLogger() {
+	ctx, cancelFn := context.WithCancel(context.Background())
```

Comment:
```
type Target struct {
	status int32

	// Channel of log entries
	logCh chan interface{}

	config Config
}
```

... at end of `func (h *Target) Init() error {`

```
	h.status = 1
	go h.startHTTPLogger()
	return nil
}
```

```
// Send log message 'e' to http target.
func (h *Target) Send(entry interface{}, errKind string) error {
	if atomic.LoadInt32(&h.status) == 0 {
		// Channel was closed or used before init.
		return nil
	}
	select {
	case h.logCh <- entry:
	default:
		// log channel is full, do not wait and return
		// an error immediately to the caller
		return errors.New("log buffer full")
	}

	return nil
}
```

You can also remove the `	go func() {` from `startHTTPLogger`, since it is already in a goroutine when started.

Reply:
The only caveat about this is that events will be sent out-of-order.

If you do not want that, add a `sync.WaitGroup` to `Target` and have `Cancel()` wait for everything to be flushed.

Reply:
Thanks @klauspost ! Have made the changes suggested by you, and also added `sync.WaitGroup`. PTAL.